eventmesh/ApiServerEventMesh.js

```diff
@@ -0,0 +1,303 @@
+        .v1alpha1.namespaces('default')[opts.annotationType].post({
+          body: initialResource
+        }))
+      .then(() => apiserver.apis[`${opts.annotationName}.servicefabrik.io`]
+        .v1alpha1.namespaces('default')[opts.annotationType](opts.annotationId).status.patch({
+          body: statusJson
+        }))
+  }
+  /**
+   * @params opts.resourceId
+   * @params opts.annotationName
+   * @params opts.annotationType
+   * @params opts.annotationId
+   * @params opts.value
+   */
+  updateAnnotationResult(opts) {
+    const patchedResource = {
+      "status": {
+        "response": JSON.stringify(opts.value),
+      }
+    }
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver
+        .apis[`${opts.annotationName}.servicefabrik.io`]
+        .v1alpha1.namespaces('default')[opts.annotationType](opts.annotationId)
+        .status.patch({
+          body: patchedResource
+        }))
+  }
+
+  /**
+   * @params opts.resourceId
+   * @params opts.annotationName
+   * @params opts.annotationType
+   * @params opts.annotationId
+   * @params opts.stateValue
+   */
+  updateAnnotationState(opts) {
+    assert.ok(opts.resourceId, `Property 'resourceId' is required to update annotation state`);
+    assert.ok(opts.annotationName, `Property 'annotationName' is required to update annotation state`);
+    assert.ok(opts.annotationType, `Property 'annotationType' is required to update annotation state`);
+    assert.ok(opts.annotationId, `Property 'annotationId' is required to update annotation state`);
+    assert.ok(opts.stateValue, `Property 'stateValue' is required to update annotation state`);
+    const patchedResource = {
+      "status": {
+        "state": opts.stateValue
+      }
+    }
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver
+        .apis[`${opts.annotationName}.servicefabrik.io`]
+        .v1alpha1.namespaces('default')[opts.annotationType](opts.annotationId)
+        .status.patch({
+          body: patchedResource
+        }))
+  }
+
+  /**
+   * @params opts.resourceId
+   * @params opts.annotationName
+   * @params opts.annotationType
+   * @params opts.value
+   */
+  updateLastAnnotation(opts) {
+    const patchedResource = {}
+    patchedResource["metadata"] = {}
+    patchedResource.metadata["labels"] = {}
+    patchedResource.metadata.labels[`last_${opts.annotationName}_${opts.annotationType}`] = opts.value
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver
+        .apis['deployment.servicefabrik.io']
+        .v1alpha1.namespaces('default')["directors"](opts.resourceId)
+        .patch({
+          body: patchedResource
+        }))
+  }
+
+  /**
+   * @params opts.resourceId
+   * @params opts.annotationName
+   * @params opts.annotationType
+   */
+  getLastAnnotation(opts) {
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver
+        .apis[`deployment.servicefabrik.io`]
```

Comment:
constants

`'default'` namespace
`.servicefabrik.io` host

Also, how do we get `directors` here?

Reply:
Done!